sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoderSuite.scala

```diff
@@ -239,6 +240,30 @@ class ExpressionEncoderSuite extends SparkFunSuite {
     ExpressionEncoder.tuple(intEnc, ExpressionEncoder.tuple(intEnc, longEnc))
   }
 
+  test("Option in Array") {
+    val data = (Seq(Some(1), Some(2), None), "Option in array")
+
+    val schema = ScalaReflection.schemaFor[Tuple2[Seq[Option[Int]], String]]
+      .dataType.asInstanceOf[StructType]
+    val attributeSeq = schema.toAttributes
+    val dataEncoder = encoderFor[Tuple2[Seq[Option[Int]], String]]
+    val boundEncoder = dataEncoder.resolve(attributeSeq, outers).bind(attributeSeq)
+    assert(boundEncoder.fromRow(boundEncoder.toRow(data)) === (
+      Seq(Some(1), Some(2), null), "Option in array"))
+  }
+
+  test("Option in Map") {
+    val data = (Map(1 -> Some(10L), 2 -> Some(20L), 3 -> None), "Option in map")
+
+    val schema = ScalaReflection.schemaFor[Tuple2[Map[Int, Option[Long]], String]]
+      .dataType.asInstanceOf[StructType]
+    val attributeSeq = schema.toAttributes
+    val dataEncoder = encoderFor[Tuple2[Map[Int, Option[Long]], String]]
+    val boundEncoder = dataEncoder.resolve(attributeSeq, outers).bind(attributeSeq)
+    assert(boundEncoder.fromRow(boundEncoder.toRow(data)) === (
+      Map(1 -> Some(10L), 2 -> Some(20L), 3 -> null), "Option in map"))
+  }
```

Comment:
I am not sure about this. Because you can see at [ScalaReflectionRelationSuite.scala#L128](https://github.com/apache/spark/blob/d83c2f9f0b08d6d5d369d9fae04cdb15448e7f0d/sql/core/src/test/scala/org/apache/spark/sql/ScalaReflectionRelationSuite.scala#L128) where this test is originally coming from, looks like it is expected to have `Seq(Some(1), null)` converted for `Seq(Some(1), None)`.